lib/Backends/Habana/Habana.h

```diff
@@ -108,7 +108,7 @@ class HabanaIOBufferPool {
   uint8_t *buffer_;
   /// The number of buffers in the pool.
   unsigned numBuffers_{kDefaultNumBuffers};
-  static constexpr unsigned kDefaultNumBuffers{3};
+  static constexpr unsigned kDefaultNumBuffers{10};
```

Comment:
Why is this being increased? I have some suspicions but it would be nice to include the reason in the commit/PR message.

Reply:
This number determines the size of the pipeline (in batches) allowed on the Habana device: functionally rate limiting calls to synEnqueue. I noticed 3 was too small (was causing the runner threads to block consistently - see first trace above for getIOBuffer). 10 is as arbitrary a number as 3, but higher is better as it will result in less context switches (at the cost of memory).